pkg/check/zones.go

```diff
@@ -0,0 +1,189 @@
+package check
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/vmware/govmomi/vim25/mo"
+	vim25types "github.com/vmware/govmomi/vim25/types"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/klog/v2"
+	"regexp"
+	"strings"
+)
+
+type validationContext struct {
+	reference           vim25types.ManagedObjectReference
+	regionTagCategoryID string
+	zoneTagCategoryID   string
+}
+
+const (
+	// TagCategoryRegion the tag category associated with regions.
+	TagCategoryRegion = "openshift-region"
+
+	// TagCategoryZone the tag category associated with zones.
+	TagCategoryZone = "openshift-zone"
+)
+
+// validateTagCategories will verify that the categories exist for the region
+// and zone tags.
+func validateTagCategories(ctx *CheckContext) (string, string, error) {
+	klog.V(4).Info("Validating tag categories...")
+	categories, err := getCategories(ctx)
+	if err != nil {
+		return "", "", err
+	}
+
+	regionTagCategoryID := ""
+	zoneTagCategoryID := ""
+	for _, category := range categories {
+		switch category.Name {
+		case TagCategoryRegion:
+			regionTagCategoryID = category.ID
+		case TagCategoryZone:
+			zoneTagCategoryID = category.ID
+		}
+		if len(zoneTagCategoryID) > 0 && len(regionTagCategoryID) > 0 {
+			break
+		}
+	}
+
+	klog.V(4).Infof("Determined RegionTagCategoryID='%s' and ZoneTagCategoryID='%s'",
+		regionTagCategoryID, zoneTagCategoryID)
+
+	if len(zoneTagCategoryID) == 0 || len(regionTagCategoryID) == 0 {
+		return "", "", fmt.Errorf("tag categories openshift-zone and openshift-region must be created")
+	}
+	return regionTagCategoryID, zoneTagCategoryID, nil
+}
+
+// validateTagAttachment will attempt to validate if region and zone tag is present
+// in the ancestry of the ManagedObjectRefence that is provided in the
+// validationContext.
+func validateTagAttachment(ctx *CheckContext, vctx validationContext) error {
+	klog.V(2).Infof("Validating tags for %s.", vctx.reference)
+	referencesToCheck := []mo.Reference{vctx.reference}
+	ancestors, err := getAncestors(ctx, vctx.reference)
```

Comment:
This value of `err` is never used. Again this is coming from gopls and hence some kind of go checker will catch this, so lets fix this kind of warnings/errors. In this case, it might be worth returning an error from the function, if we can't fetch ancestor?

Reply:
Good catch.  I had one there, and not sure why it disappeared.  I will re-add the error return block.